connector/github/github.go

```diff
@@ -313,55 +313,36 @@ func (c *githubConnector) listGroups(ctx context.Context, client *http.Client, u
 	for _, org := range c.orgs {
 		inOrg, err = c.userInOrg(ctx, client, userName, org.Name)
 		if err != nil {
-			c.logger.Errorf("user in org: %s", err)
-			continue
+			return groups, err
 		}
 		if !inOrg {
 			c.logger.Debugf("github: user %q not in org %q", userName, org.Name)
 			continue
 		}
 
 		if teams, err = c.teams(ctx, client, org.Name); err != nil {
-			c.logger.Errorf("teams: %s", err)
-			continue
+			return groups, err
 		}
 
-		// User is in at least in one org. No team restrictions are imposed from
-		// config. We add all the teams in the organization the user is a member of
-		// to 'groups' to preserve legacy behavior.
-		if len(org.Teams) == 0 {
-			groups = append(groups, joinOrgToTeams(org.Name, teams)...)
+		// Orgs might have the same team names. We append orgPrefix to team name,
+		// i.e. "org:team", to make team names unique across orgs.
+		orgPrefix := org.Name + ":"
+		// User is in at least in one org. If team restrictions are present in config,
+		// ensure user is a member of at least one team.
+		if len(org.Teams) > 0 {
+			for _, teamName := range filterTeams(teams, org.Teams) {
+				groups = append(groups, orgPrefix+teamName)
+				c.logger.Debugf("github: user %q in org %q team %q", userName, org.Name, teamName)
+			}
+		} else { // No team restrictions. Append "org:" to groups
```

Comment:
I really dislike this. Even if we're going to do this, we should always add the "(org):" or even just "(org)"